tests/bson-corpus/binary-valid-014.phpt

```diff
@@ -0,0 +1,43 @@
+--TEST--
+Binary type: $type query operator (conflicts with legacy $binary form with $type field)
+--DESCRIPTION--
+Generated by scripts/convert-bson-corpus-tests.php
+
+DO NOT EDIT THIS FILE
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/basic.inc';
+
+$canonicalBson = hex2bin('180000000378001000000010247479706500020000000000');
+$canonicalExtJson = '{"x" : { "$type" : {"$numberInt": "2"}}}';
+
+// Canonical BSON -> Native -> Canonical BSON
+echo bin2hex(fromPHP(toPHP($canonicalBson))), "\n";
+
+// Canonical BSON -> BSON object -> Canonical BSON
+echo bin2hex((string) MongoDB\BSON\Document::fromBSON($canonicalBson)), "\n";
+
+// Canonical BSON -> Canonical extJSON
+echo json_canonicalize(toCanonicalExtendedJSON($canonicalBson)), "\n";
```

Comment:
I think `json_canonicalize` is not needed for the tested data.

I've seen the `json_canonicalize` function have a patch for PHP < 7.1 that could be removed.

Reply:
Created [PHPC-2320](https://jira.mongodb.org/browse/PHPC-2320) to remove the string replacement logic. The function is still relevant for `$numberDouble`.